# What should I learn? {#what-to-learn}

Modern scientific coders often operate with many coding languages and software in their workflow. But getting started they often have a primary language that they use along with a suite of concepts and that help them develop workflows and interoperate between languages and software to do their analytical research. Learning one language to do a task at hand can help you develop skills and mindsets that can be transferrable when you come upon other tasks. 
A great thing about modern computation is that there are multiple ways to do the same thing. But this can also be a really challenging thing too: what should you learn? And in which order? This chapter aims to help you find resources to learn what you need. 



<!---

Recommendations: keep a diary, 
- articulate your problem so you can google it
- look for topics that are useful and recent, community vetted (trust)
- keep a record for yourself. In many ways, this book is a record 
to help you as a researcher identify which skills you need to develop and be deliberate about *why*.

Different learner styles, more directed self-taught less wasted time and wheels reinvented.

<!---We've often never had these conversations and this book tries to provide an opportunity.

Quotes: 
- " Part of learning how to be an effective computational biologist is to learn how to learn efficiently."
- "I think it’s really hard for new people to jump in. The learning curves can be steep, and it’s hard for them to judge how much should they expect "


--->



---
## Learn a new language
Introductory tutorials will help you install software and get comfortable using it. They will also and help you develop the mindset to seek out existing code that works (including your own!) and adapt it for your analyses. 

<!---There are often stereotypes about programming languages that --->

To get a sense of coding in Python...\@ref(python-sense)

You want to learn: 

- **R** ...\@ref(r-intro)  
- **Python**
  - with no previous coding experience... \@ref(python-first)
  - with previous coding experience...\@ref(python-second) 
- **The Command Line**...  
- You are not sure or need to use multiple languages together...\@ref(workflow-overview)  

## Image analysis 

<!---(single cell)--->

You want to learn: 

- **CellProfiler**...  <https://cellprofiler.org/>
- **ImageJ**...
  - with no previous coding experience...\@ref(imagej-intro)
  - to write automated macros and plugins...\@ref(imagej-macros)
- **QuPath**...  <https://qupath.github.io/>
- **Napari**... <https://napari.org/tutorials/> - Nick Sofroniew
<!---but right now if you're working in Python and doing image analysis you can get going from here--->
<!--- - streamline your current workflow... --- with R/Python...--->

## RNA Sequencing

- Molly Hammell software
- Gene Yeo UCSD course


## Project Management & Workflows

- Google Drive / Box / Dropbox...
- AWS...
- Project-oriented workflows....\@ref(proj-oriented)
- R Markdown...\@ref(rmarkdown)
- Jupyter Notebooks...
- GitHub Issues...\@ref(gh-issues)
- evcouplings pipeline...\@ref(ndcn-evcouplings)
- text editors (could be its own chapter)

## GitHub

- intro with R ...\@ref(r-intro)
- walking back changes with R...\@ref(gh-fix)
- Issues...\@ref(gh-issues)

<!---
## Manuscripts

- should I use latex? Overleaf
- Rmd
--->

## Data Sharing
Platforms for data sharing <!---Gene Yeo CRED portal as an example:) ) https://www.dropbox.com/s/9gvep1mkqxsj7g1/CRED_slides.pptx?dl=0)

-->

## Machine Learning (ML)

<!--- tensorflow, fast.ai, Kelly Brock--->

- Intro to ML conceptually ...
- Intro ML in R....\@ref(rstats-ml)  
- Intro ML in Python....

## Plotting 
*should this be a section or page?*


## Publishing
*create its own page* 

- GitHub for Publishing...\@ref(gh-issues)

<!---manuscripts and preprints --->


## Containers
- Docker ...
