# What should I learn? {#what-to-learn}

A great thing about modern computation is that there are multiple ways to do the same thing. But this can also be a really challenging thing about modern computing: what should you learn? And in which order?

Modern scientific coders use don't just know a single language or tool, they know a suite of tools and concepts that help them develop workflows and interoperate between software to do their analytical research. 

*Preamble...also modern coders use collaborative version control software like GitHub. Workflows and interoperating between tools and software is really critical.*

*Add opening narratives/personas: you are running other people's scripts but need a workflow, you have your own data but don't know how to analyze it...*

<!---
Recommendations: keep a diary, 
- articulate your problem so you can google it
- look for topics that are useful and recent, community vetted (trust)
- keep a record for yourself. In many ways, this book is a record 
to help you as a researcher identify which skills you need to develop and be deliberate about *why*.

Different learner styles, more directed self-taught less wasted time and wheels reinvented.

<!---We've often never had these conversations and this book tries to provide an opportunity.

Quotes: 
- " Part of learning how to be an effective computational biologist is to learn how to learn efficiently."
- "I think it’s really hard for new people to jump in. The learning curves can be steep, and it’s hard for them to judge how much should they expect "


--->



---
## Intro coding
Introductory tutorials will help you install software and get comfortable using it. They will also and help you develop the mindset to seek out existing code that works (including your own!) and adapt it for your analyses. 

- You want to learn R ...\@ref(r-intro)  
- You want to learn Python...  
- You want to learn Unix...  
- You are not sure or need to use multiple languages together...\@ref(workflow-overview)  

## Image analysis 
(create a separate chapter)

(single cell)

- You want to learn CellProfiler...  <https://cellprofiler.org/>
- You want to learn ImageJ...  <https://imagej.net/Welcome> <!---https://www.protocols.io/view/imagej-plate-count-protocol-f2mbqc6--->
- You want to learn QuPath...  <https://qupath.github.io/>
- You want to learn Napari... <https://napari.org/tutorials/> - Nick Sofroniew
<!---but right now if you're working in Python and doing image analysis you can get going from here--->
- You want to use streamline your current workflow... <!--- with R/Python...--->

## RNA Sequencing

- Molly Hammell software
- Gene Yeo UCSD course


## Project Management & Workflows

- Google Drive / Box / Dropbox...
- AWS...
- Project-oriented workflows....\@ref(proj-oriented)
- R Markdown...\@ref(rmarkdown)
- Jupyter Notebooks...
- GitHub Issues...\@ref(gh-issues)
- evcouplings pipeline...\@ref(ndcn-evcouplings)
- text editors (could be its own chapter)

## GitHub

- intro with R ...\@ref(r-intro)
- walking back changes with R...\@ref(gh-fix)
- Issues...\@ref(gh-issues)

<!---
## Manuscripts

- should I use latex? Overleaf
- Rmd
--->

## Data Sharing
Platforms for data sharing <!---Gene Yeo CRED portal as an example:) ) https://www.dropbox.com/s/9gvep1mkqxsj7g1/CRED_slides.pptx?dl=0)


## Machine Learning (ML)

<!--- tensorflow, fast.ai, Kelly Brock--->

- Intro to ML conceptually ...
- Intro ML in R....\@ref(rstats-ml)  
- Intro ML in Python....

## Plotting 
*should this be a section or page?*


## Publishing
*create its own page* 

- GitHub for Publishing...\@ref(gh-issues)

<!---manuscripts and preprints --->


## Containers
- Docker ...
